execution_engine/src/core/engine_state/prune.rs

```diff
@@ -0,0 +1,44 @@
+//! Support for pruning leaf nodes from the merkle trie.
+use casper_hashing::Digest;
+use casper_types::Key;
+
+/// Represents a successfully performed prune.
```

Comment:
```suggestion
/// The result of performing a prune.
```

Reply:
Did this (just to let you know `UpgradeResult` etc they have the same `Represents a successfully performed XYZ` comments)

Reply:
Maybe worth updating their comments too then if they say they represent success, but contain variants which look like failure cases?  (Not in this PR - I just mean some time in the future).